### PR TITLE
Add huwelijk

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: Service/ZgwToVrijbrpService.php [hunk 1]
```diff
@@ -394,6 +662,8 @@ private function synchronizeTemp(Synchronization $synchronization, array $object
     {
         $objectString = $this->syncService->getObjectString($objectArray);
 
+        $this->logger->debug("SynchronizeTemp with objectString: $objectString");
+
         $this->logger->info('Sending message with body '.$objectString);
```
Comment (WilcoLouwerse) [inline Service/ZgwToVrijbrpService.php]: log twice?

File: Service/ZgwToVrijbrpService.php [hunk 2]
```diff
@@ -243,13 +356,165 @@ private function getBirthProperties(array $zgw, array $output): array
...
+        // If count($keys) == 1
+        // Todo: This with only 1 key is only used for bsn, so could be removed if we don't use it for bsn anymore
+        if (isset($zaakEigenschappen['partner1'][$keys[0]]) === true) {
+            $zaakEigenschappen['partner2'][$keys[0]] = $eigenschap->getValue('waarde');
+            return;
+        }
+        $zaakEigenschappen['partner1'][$keys[0]] = $eigenschap->getValue('waarde');
+    }//end getCommitmentPartnerEigenschap()
+
+    private function flattenEigenschappen(array $eigenschappen): array
```
Comment (WilcoLouwerse) [inline Service/ZgwToVrijbrpService.php]: Remove this function and use getZaakEigenschappen() instead?

File: Service/ZgwToVrijbrpService.php [hunk 3]
```diff
@@ -243,13 +356,165 @@ private function getBirthProperties(array $zgw, array $output): array
...
+     */
+    private function getCommitmentZaakEigenschappen(ObjectEntity $zaakObjectEntity, array $properties): array
+    {
+        $zaakEigenschappen = [];
+        $eigenschappen = $this->getZaakEigenschappen($zaakObjectEntity, ['all']);
+        $tempCountForBsn = 1; // Todo: fix this bsn shizzle
+        foreach ($zaakObjectEntity->getValue('eigenschappen') as $eigenschap) {
+            switch ($eigenschap->getValue('naam')) {
+                case 'inp.bsn':
+                    // Todo: fix this bsn shizzle:
```
Comment (WilcoLouwerse) [inline Service/ZgwToVrijbrpService.php:394]: Does this work, do we still need this?

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: Service/ZgwToVrijbrpService.php (hunk 1 above)
Comment (rjzondervan) [reply Service/ZgwToVrijbrpService.php]: Interesting merge thing, will fix

File: Service/ZgwToVrijbrpService.php (hunk 2 above)
Comment (rjzondervan) [reply Service/ZgwToVrijbrpService.php]: getZaakEigenschappen resulted in 504's with witnesses

File: Service/ZgwToVrijbrpService.php (hunk 3 above)
Comment (rjzondervan) [reply Service/ZgwToVrijbrpService.php:394]: It works, and yes, we still need it

File: Service/ZgwToVrijbrpService.php (hunk 2 above)
Comment (WilcoLouwerse) [reply Service/ZgwToVrijbrpService.php]: But you aren't using this function flattenEigenschappen(), you are using getZaakEigenschappen()

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**